### PR TITLE
Replace sqs with redis, drop pycurl.

### PR DESCRIPTION
**Pull request by katherine-black**
#433 

Depends on https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/31835 merging first.

### CODE REVIEW

File: cloudigrade/config/settings/base.py
Comment (infinitewarp) [inline cloudigrade/config/settings/base.py:401]: I also mentioned something about this in the app-interface MR, but just to make sure I cover all the bases…

`inMemoryDb` is somewhat worrying to see since we need our queues to be persistent across restarts. Does this mean something other than what it looks like at first glance?

File: deployment/clowdapp.yaml
Comment (infinitewarp) [inline deployment/clowdapp.yaml:13]: Ditto earlier comments/questions about `inMemoryDb`.

**Comment by infinitewarp** [comment]
Other than my questions (hopefully just my misunderstanding) about the in-memory DB settings,

File: cloudigrade/config/settings/base.py
Comment (katherine-black) [reply cloudigrade/config/settings/base.py:401]: Yea so this is just a super confusing component on bonfire/clowder/something. Its only in memory in name, [the docs touch on there actually being two backing mechanisms](https://redhatinsights.github.io/clowder/clowder/dev/providers/inmemorydb.html), in the case of ephemeral its a pod running redis with a pvc. in stage and prod its our deployed app-sre elasticache clusters. Hopefully this helps clear it up.

File: deployment/clowdapp.yaml
Comment (katherine-black) [reply deployment/clowdapp.yaml:13]: replied above ^

File: cloudigrade/config/settings/base.py
Comment (infinitewarp) [reply cloudigrade/config/settings/base.py:401]: Thanks for the explanation! 🙂 That seems reasonable. I'm going to ask around elsewhere about those docs because, yeah, that's heckin' confusing. 😕

**Review by infinitewarp — APPROVED** [review]

**Review by infinitewarp — CHANGES_REQUESTED** [review]
After stepping away to think a bit more, I realize I let my desire to purge pycurl get the better of me 😅, and I'd actually like to see this in action a bit more before merging. I'm slapping this comment here just to block merge until then.

Though I trust Celery's integration with Redis should work fine in general, I'd like to see it running with some of our tasks executing in OpenShift, to see the Redis deployment in this specific configuration successfully recover unread/unacknowledged messages after a restart, and to see what methods and tools we have at our disposal to monitor, investigate, and diagnose potential issues. If you've already done some of this investigation or have thoughts and want to give a demo, that would be great. Otherwise I'll pull this branch and experiment on my own tomorrow.

**Comment by katherine-black** [comment]
So far its been through a few happy path scenarios in ephemeral. I was hoping the smoke job would start working again so I could throw that at it next. As for performance, its a concern for me in stage and prod, those elasticache instances will be in an AWS account we do not have access to, so having metrics for that is crucial. I was actually hoping you'd have some idea in that space with your recent prometheus and grafana work, but we may need tor reach out to app-sre for more info. 👍 for taking infra changes slowly though :)

**Comment by katherine-black** [comment]
> Do you have any good tips for checking the state of the queues? I mean in terms of their content, not necessarily performance. Maybe this is where we can start using Flower in the near future since Redis should have better support?

Yup, especially since we also now have the results being placed back on the queue, Flower is indeed what I had in mind for the next step.

**Comment by katherine-black** [comment]
@infinitewarp now that the app-sre PR was merged, I found the grafana pages for the cluster stats - 

[prod #1](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-prod&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-001&var-cachenodeid=0001)
[prod #2](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-prod&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-002&var-cachenodeid=0001)
[stage #1](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-stage&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-001&var-cachenodeid=0001)
[stage #2](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-stage&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-002&var-cachenodeid=0001)

**Comment by infinitewarp** [comment]
> @infinitewarp now that the app-sre PR was merged, I found the grafana pages for the cluster stats -
> 
> [prod #1](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-prod&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-001&var-cachenodeid=0001) [prod #2](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-prod&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-002&var-cachenodeid=0001) [stage #1](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-stage&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-001&var-cachenodeid=0001) [stage #2](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-stage&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-002&var-cachenodeid=0001)

@katherine-black I'm getting "Dashboard not found" for all of those links now. Do you know if they're still available? Moved or renamed somewhere?

**Comment by katherine-black** [comment]
> > @infinitewarp now that the app-sre PR was merged, I found the grafana pages for the cluster stats -
> > [prod #1](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-prod&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-001&var-cachenodeid=0001) [prod #2](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-prod&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-002&var-cachenodeid=0001) [stage #1](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-stage&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-001&var-cachenodeid=0001) [stage #2](https://grafana.app-sre.devshift.net/d/iWGjxRbnz/aws-elasticache-redis?orgId=1&var-datasource=AWS%20insights-stage&var-region=default&var-cacheclusterId=cloudigrade-redis-clowder-002&var-cachenodeid=0001)
> 
> @katherine-black I'm getting "Dashboard not found" for all of those links now. Do you know if they're still available? Moved or renamed somewhere?

Yea the URL chagned for some reason, you can find it by just searching for elasticache in grafana. https://grafana.app-sre.devshift.net/d/iM4meNbnz/aws-elasticache-redis?orgId=1 Just select the insights datasources and we'll be there.

**Review by infinitewarp — APPROVED** [review]
I did some local testing in an ephemeral namespace deployment, and that alleviated some of my concerns. Let's see how this behaves for real!

**Pull request merged**
